File: src/main/java/org/cimsbioko/scripting/MessagesScriptable.java
package org.cimsbioko.scripting;

import org.mozilla.javascript.Scriptable;

public class MessagesScriptable implements Scriptable {

    private final JsConfig config;

    MessagesScriptable(JsConfig config) {
        this.config = config;
    }

    @Override
    public String getClassName() {
        return "MessagesScriptable";
    }

    @Override
    public Object get(String name, Scriptable start) {
        return config.getString(name);
    }

    @Override
    public Object get(int index, Scriptable start) {
        return null;
    }

    @Override
    public boolean has(String name, Scriptable start) {
        return true;
    }

    @Override
    public boolean has(int index, Scriptable start) {
        return false;
    }

    @Override
    public void put(String name, Scriptable start, Object value) {

    }

    @Override
    public void put(int index, Scriptable start, Object value) {

    }

    @Override
    public void delete(String name) {

    }

    @Override
    public void delete(int index) {

    }

    @Override
    public Scriptable getPrototype() {
        return null;
    }

    @Override
    public void setPrototype(Scriptable prototype) {

    }

    @Override
    public Scriptable getParentScope() {
        return null;
    }

    @Override
    public void setParentScope(Scriptable parent) {
    }

    @Override
    public Object[] getIds() {
        return null;
    }

    @Override
    public Object getDefaultValue(Class<?> hint) {
        return null;
    }

    @Override
    public boolean hasInstance(Scriptable instance) {
        return false;
    }


}

File: src/main/java/org/cimsbioko/scripting/MessagesScriptable.kt
package org.cimsbioko.scripting

import org.mozilla.javascript.Scriptable

class MessagesScriptable internal constructor(private val config: JsConfig) : Scriptable {
    override fun getClassName() = "MessagesScriptable"
    override fun get(name: String, start: Scriptable): String = config.getString(name)
    override fun get(index: Int, start: Scriptable): Any? = null
    override fun has(name: String, start: Scriptable) = true
    override fun has(index: Int, start: Scriptable): Boolean = false
    override fun put(name: String, start: Scriptable, value: Any) {}
    override fun put(index: Int, start: Scriptable, value: Any) {}
    override fun delete(name: String) {}
    override fun delete(index: Int) {}
    override fun getPrototype(): Scriptable? = null
    override fun setPrototype(prototype: Scriptable) {}
    override fun getParentScope(): Scriptable? = null
    override fun setParentScope(parent: Scriptable) {}
    override fun getIds(): Array<Any>? = null
    override fun getDefaultValue(hint: Class<*>?): Any? = null
    override fun hasInstance(instance: Scriptable): Boolean = false
}